2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_setup.py:_flush():81] Current SDK version is 0.20.1
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_setup.py:_flush():81] Configure stats pid to 41927
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_setup.py:_flush():81] Loading settings from /home/ryan/.config/wandb/settings
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_setup.py:_flush():81] Loading settings from /home/ryan/toy_mazero/wandb/settings
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_setup.py:_flush():81] Loading settings from environment variables
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_init.py:setup_run_log_directory():703] Logging user logs to /home/ryan/toy_mazero/wandb/run-20250629_165004-tmoaomhy/logs/debug.log
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_init.py:setup_run_log_directory():704] Logging internal logs to /home/ryan/toy_mazero/wandb/run-20250629_165004-tmoaomhy/logs/debug-internal.log
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_init.py:init():831] calling init triggers
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_init.py:init():836] wandb.init called with sweep_config: {}
config: {'train': TrainConfig(env_name='MPE_simple_spread_v3', num_agents=3, num_episodes=500000, warmup_episodes=1000, log_interval=100, num_actors=6, max_episode_steps=25, replay_buffer_size=20000, replay_buffer_alpha=0.6, replay_buffer_beta_start=0.4, replay_buffer_beta_frames=100000, batch_size=256, learning_rate=0.0001, param_update_interval=10, end_lr_factor=0.1, lr_warmup_steps=5000, value_loss_coefficient=1.0, consistency_coeff=1.0, gradient_clip_norm=5.0, unroll_steps=5, n_step=1, discount_gamma=0.99), 'model': ModelConfig(hidden_state_size=128, value_support_size=300, reward_support_size=300, fc_representation_layers=(128,), fc_dynamic_layers=(128,), fc_reward_layers=(32,), fc_value_layers=(32,), fc_policy_layers=(32,), attention_type='none', attention_layers=3, attention_heads=4, dropout_rate=0.1, proj_hid=256, proj_out=256, pred_hid=64, pred_out=256), 'mcts': MCTSConfig(planner_mode='joint', num_simulations=50, num_joint_samples=16, max_depth_gumbel_search=10, num_gumbel_samples=10), '_wandb': {}}
2025-06-29 16:50:04,053 INFO    MainThread:41927 [wandb_init.py:init():872] starting backend
2025-06-29 16:50:04,262 INFO    MainThread:41927 [wandb_init.py:init():875] sending inform_init request
2025-06-29 16:50:04,266 INFO    MainThread:41927 [wandb_init.py:init():883] backend started and connected
2025-06-29 16:50:04,267 INFO    MainThread:41927 [wandb_init.py:init():956] updated telemetry
2025-06-29 16:50:04,272 INFO    MainThread:41927 [wandb_init.py:init():980] communicating run to backend with 90.0 second timeout
2025-06-29 16:50:04,759 INFO    MainThread:41927 [wandb_init.py:init():1032] starting run threads in backend
2025-06-29 16:50:04,830 INFO    MainThread:41927 [wandb_run.py:_console_start():2453] atexit reg
2025-06-29 16:50:04,830 INFO    MainThread:41927 [wandb_run.py:_redirect():2301] redirect: wrap_raw
2025-06-29 16:50:04,830 INFO    MainThread:41927 [wandb_run.py:_redirect():2370] Wrapping output streams.
2025-06-29 16:50:04,831 INFO    MainThread:41927 [wandb_run.py:_redirect():2393] Redirects installed.
2025-06-29 16:50:04,832 INFO    MainThread:41927 [wandb_init.py:init():1078] run started, returning control to user process
2025-06-29 16:50:27,837 INFO    MsgRouterThr:41927 [mailbox.py:close():129] [no run ID] Closing mailbox, abandoning 1 handles.
2025-06-29 16:50:28,898 ERROR   MainThread:41927 [redirect.py:_on_write():664] [no run ID] error in stderr callback
Traceback (most recent call last):
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/redirect.py", line 662, in _on_write
    cb(written_data)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 2380, in <lambda>
    lambda data: self._console_raw_callback("stderr", data),
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 391, in wrapper
    return func(self, *args, **kwargs)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 449, in wrapper_fn
    return func(self, *args, **kwargs)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 1561, in _console_raw_callback
    self._backend.interface.publish_output_raw(name, data)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/interface/interface.py", line 772, in publish_output_raw
    self._publish_output_raw(o)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/interface/interface_shared.py", line 38, in _publish_output_raw
    self._publish(rec)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/interface/interface_sock.py", line 39, in _publish
    self._sock_client.send_record_publish(record)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 174, in send_record_publish
    self.send_server_request(server_req)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 154, in send_server_request
    self._send_message(msg)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 151, in _send_message
    self._sendall_with_error_handle(header + data)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 130, in _sendall_with_error_handle
    sent = self._sock.send(data)
BrokenPipeError: [Errno 32] Broken pipe
2025-06-29 16:50:28,908 ERROR   MainThread:41927 [redirect.py:_on_write():664] [no run ID] error in stderr callback
Traceback (most recent call last):
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/redirect.py", line 662, in _on_write
    cb(written_data)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 2380, in <lambda>
    lambda data: self._console_raw_callback("stderr", data),
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 391, in wrapper
    return func(self, *args, **kwargs)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 449, in wrapper_fn
    return func(self, *args, **kwargs)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 1561, in _console_raw_callback
    self._backend.interface.publish_output_raw(name, data)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/interface/interface.py", line 772, in publish_output_raw
    self._publish_output_raw(o)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/interface/interface_shared.py", line 38, in _publish_output_raw
    self._publish(rec)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/interface/interface_sock.py", line 39, in _publish
    self._sock_client.send_record_publish(record)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 174, in send_record_publish
    self.send_server_request(server_req)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 154, in send_server_request
    self._send_message(msg)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 151, in _send_message
    self._sendall_with_error_handle(header + data)
  File "/home/ryan/miniconda3/envs/jaxmarl_env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 130, in _sendall_with_error_handle
    sent = self._sock.send(data)
BrokenPipeError: [Errno 32] Broken pipe
